aws/asg.go

```diff
@@ -103,7 +103,7 @@ func getOwnedAutoScalingGroups(service autoscalingiface.AutoScalingAPI, clusterI
 					value := aws.StringValue(td.Value)
 					tags[key] = value
 
-					if key == clusterIDTag && value == resourceLifecycleOwned {
+					if key == clusterIDTag && value == resourceLifecycleOwned && asgNameMap[name] {
```

Comment:
This unfortunately breaks what the other PR introducing the regression fixed. Which is find ASGs which are managed but currently have 0 running instances.

It seems to me these two features are somewhat conflicting. At least in the way we are trying to fix them right now.

I think the only right way is to look at ASG tags and see if they match the filter.

Reply:
Hmmmm. I was a little worried about that.... I had taken a couple of runs at adding tag filters to ASG, but I really didn't like the shape of what was coming out (largely duplicating functions used for the instance filtering, but which I couldn't re-use due to slightly different types in the AWS APIs), and came back to this.

I'll keep pinning to 0.8.3 and see if I have time to try again soon.